nx-X11/programs/Xserver/hw/nxagent/Args.c

```diff
@@ -1545,45 +1545,138 @@ static void nxagentParseOptions(char *name, char *value)
   free(argv[0]);
 }
 
+/*
+  little helper function to handle quoted/parenthesized/bracketed options
+
+  string      the option string up to the first comma (will be modified!)
+  fullstring  the full option string to the end (will not be modified)
+
+  If the function finds it needs more characters it will take
+  those from fullstring. It will return the position in fullstring
+  after the comma following the current processed option
+
+*/
+static char *processoption(char *string, char *fullstring)
+{
+  const char startchars[] = "'\"|{([<";
+  const char endchars[]   = "'\"|})]>";
+
+  char *delimiter = NULL;
+  char *value = NULL;
+  char *realvalue = NULL;
+  char *next = fullstring + strlen(string);
+
+  if ((delimiter = index(string, '=')))
+  {
+    *delimiter = 0;
+    value = delimiter + 1;
+    for (int i = 0; i < strlen(startchars); i++)
+    {
+      if (*value == startchars[i])
+      {
+        char *valstart = fullstring + (value - string) + 1;
+        if ((next = strchr(valstart, endchars[i])))
+        {
+          realvalue = strndup(valstart, next - valstart);
```

Comment:
An option string such as `foo="b,ar"baz` will be parsed wrongly, with only `b,ar` passed as the value. It's especially problematic that the remainder `baz` will be handled as a key in the next iteration.

I guess that more work is necessary here.

And it can get even more complex, think of a string like `foo="b,ar"b"az"` as well. We actually need to scan for "quotes" multiple times until we either hit another comma or a `NULL` character.

Reply:
No. Quoting characters are only allowed at the beginning and at the end of the string. Everything else is considered unsupported. and will/can produce unexpected results.

However, what about extending that a bit by only excepting the end character it it is directly followed by ',' or '\0'?

Reply:
That at least prohibits unexpected/wrong further parsing and looks correct. I guess that's good enough under the constraint you mentioned.